### PR TITLE
store/localstore: evaluate where before get row.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
@BusyJay PTAL

**Comment by zimulala** [comment]
LGTM

**Comment by BusyJay** [comment]
LGTM

File: store/localstore/local_region.go [hunk 1]
```diff
@@ -73,12 +89,43 @@ func (rs *localRegion) Handle(req *regionRequest) (*regionResponse, error) {
 	return resp, nil
 }
 
-func (rs *localRegion) getRowsFromSelectReq(txn kv.Transaction, sel *tipb.SelectRequest) ([]*tipb.Row, error) {
-	kvRanges := rs.extractKVRanges(sel.TableInfo.GetTableId(), 0, sel.Ranges)
-	eval := &xeval.Evaluator{Row: make(map[int64]types.Datum)}
+func collectColumnsInWhere(expr *tipb.Expr, ctx *selectContext) error {
+	if expr == nil {
```
Comment (c4pt0r) [inline store/localstore/local_region.go]: I think expr can't be nil. Is the check necessary?

**Comment by c4pt0r** [comment]
Add some test?

**Comment by coocood** [comment]
@c4pt0r It's a refactor, the old tests covered the new code.

**Pull request merged**